mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/modules/CassandraMailboxModule.java

```diff
@@ -42,7 +42,8 @@
         .comment("Holds the mailboxes information.")
         .options(options -> options
             .caching(SchemaBuilder.KeyCaching.ALL,
-                SchemaBuilder.rows(CassandraConstants.DEFAULT_CACHED_ROW_PER_PARTITION)))
+                SchemaBuilder.rows(CassandraConstants.DEFAULT_CACHED_ROW_PER_PARTITION))
+            .compressionOptions(SchemaBuilder.lz4().withChunkLengthInKb(8)))
```

Comment:
how did you come to choose `8` Kb ? 
is it based on BufferedReader's defaultCharBufferSize to align with memory page size or is based on the estimated size of a mailbox row ? 
(https://stackoverflow.com/questions/37404068/why-is-the-default-char-buffer-size-of-bufferedreader-8192)

a comment would be nice :D

(if the answer is size of a mailbox row, the same question applies to all the following `8` that appear in the PR)

Reply:
https://thelastpickle.com/blog/2018/08/08/compression_performance.html

TLDR:  Using out of the box settings for compression on read heavy or mixed workloads will almost certainly put unnecessary strain on your disk while hurting your read performance.

Basically I tried a couple of values and 8KB worked well.

Reply:
> TLDR: Using out of the box settings for compression on read heavy or mixed workloads will almost certainly put unnecessary strain on your disk while hurting your read performance.

I understand the logic behind the change, I challenge the arbitrary number that was chosen :D  

> Basically I tried a couple of values (4K, 8K, 16K, ...) and 8KB worked well.

Aren't you worried that it is highly dependendant on the cassandra infrastructure ? After all 640k is enough memory for everything ;p

Reply:
I do think compression is more related to the workload than the infrastructure.

That being said I'm fine keeping defaults, and requiring expertise to come and tune these settings.